.github/workflows/build-release.yaml

```diff
@@ -0,0 +1,200 @@
+            runner: matterlabs-ci-runner
+            image: matterlabs/llvm_runner:ubuntu20-llvm17-latest
+            release-suffix: linux-amd64
+          - name: "Linux ARM64"
+            runner: matterlabs-ci-runner-arm
+            image: matterlabs/llvm_runner:ubuntu20-llvm17-latest
+            release-suffix: linux-arm64
+    runs-on: ${{ matrix.runner }}
+    container:
+      image: ${{ matrix.image || '' }}
+    name: ${{ matrix.name }}
+    steps:
+
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          submodules: recursive
+
+      - name: Building the Solidity compiler
+        uses: ./.github/actions/solc-build
+        with:
+          release-suffix: ${{ matrix.release-suffix }}
+          zksync-version: ${{ needs.setup.outputs.zksync-version }}
+          solc-version: ${{ needs.setup.outputs.solc-version }}
+
+  build-windows:
+    name: Windows
+    needs: setup
+    if: needs.setup.outputs.skip-windows == 'false'
+    runs-on: windows-2019-github-hosted-16core
+    steps:
+
+      - name: Checkout source
+        uses: actions/checkout@v4
+        with:
+          submodules: recursive
+
+      - name: Install LLVM and Clang
+        uses: KyleMayes/install-llvm-action@v1
+        with:
+          version: "15.0.7"
+
+      - name: Prepare environment
+        run: |
+          git config --global user.email "runner@github.com"
+          git fetch --all
+          # try to apply win patch
+          git cherry-pick 860226e25dca397b5afd70680530963712aff050 || git cherry-pick --abort
+          # latest llvm for MSVC
+          git clone -q --depth=1 --branch=main https://github.com/zufuliu/llvm-utils.git c:\projects\llvm-utils
+          Invoke-Item "c:\projects\llvm-utils\VS2017\install.bat"
+          clang --version
+
+
+      - name: Install windows Deps
+        run: |
+          if ( -not (Test-Path "deps\boost") ) {
+            New-Item -ItemType Directory -Force -Path "deps"
+
+            Invoke-WebRequest -URI "https://github.com/Kitware/CMake/releases/download/v3.27.1/cmake-3.27.1-windows-x86_64.zip" -OutFile cmake.zip
+
+            tar -xf cmake.zip
+            mv cmake-3.27.1-windows-x86_64 "deps\cmake"
+
+            # and serving it a page that requires JavaScript.
+            Invoke-WebRequest -URI "https://boostorg.jfrog.io/artifactory/main/release/1.77.0/source/boost_1_77_0.zip" -OutFile boost.zip
+
+            tar -xf boost.zip
+            cd boost_1_77_0
+            .\bootstrap.bat --with-toolset=clang
+            .\b2 -j4 -d0 toolset=clang cxxflags="-stdlib=libc++" linkflags="-stdlib=libc++" link=static runtime-link=static variant=release threading=multi address-model=64 --prefix="..\deps\boost" --with-filesystem --with-system --with-program_options --with-test --with-regex --with-thread install
+            if ( -not $? ) { throw "Error building boost." }
+            cd ..
+          }
+
+      - name: Building the Solidity compiler
```

Comment:
Is it possible to have one step for building solc? I see a "Build solc" step in action.yml. If not, can you tell me how these 2 build steps work? Thanks!

Reply:
No, unfortunately I can't unify Windows now... for 0.8.x it works without issues with MinGW, but I had to return this ugly solution to have compatibility with older versions... The `action.yml` will be in use for Linux and MacOS, but for Windows we have to be on this boilerplate powershell :(